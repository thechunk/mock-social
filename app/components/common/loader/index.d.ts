interface ILoader {
    loading: boolean,
    style: import('react-native').StyleProp<import('react-native').ViewStyle>
}
