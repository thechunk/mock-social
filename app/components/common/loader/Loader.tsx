import * as React from 'react';
import {PureComponent} from "react";
import {ActivityIndicator, ActivityIndicatorProps, View} from "react-native";
import styles from './styles';

export default class Loader extends PureComponent<ILoader & ActivityIndicatorProps> {
    render() {
        return this.props.loading ? (
            <View style={[styles.loader, this.props.style]}>
                <ActivityIndicator size="small" {...this.props as ActivityIndicatorProps} />
            </View>
        ) : null;
    }
}