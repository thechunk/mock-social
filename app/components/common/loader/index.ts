/// <reference path='./index.d.ts' />

export {default as Loader} from './Loader';

