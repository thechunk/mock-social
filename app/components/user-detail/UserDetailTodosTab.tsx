import * as React from 'react';
import {PureComponent} from "react";
import withTodosData from "../../hoc/withTodosData";
import {TitleSubtitleListCell} from "../common/list/index";
import {Animated} from "react-native";
import Icon from 'react-native-vector-icons/Ionicons';
import styles, {inherited} from "./styles";
import {Loader} from "../common/loader/index";
import {Color} from "../../styles/global";

export class UserDetailTodosTab extends PureComponent<IUserDetailTodosTab & IWithTodosDataProps & IWithTodosDataOptionalProps,
    IUserDetailTodosTabState> {
    constructor(props: IUserDetailTodosTab & IWithTodosDataProps & IWithTodosDataOptionalProps) {
        super(props);
        this.state = {opacity: new Animated.Value(0)}
    }

    componentDidUpdate(prevProps: Readonly<IUserDetailTodosTab & IWithTodosDataProps & IWithTodosDataOptionalProps>, prevState: Readonly<{}>, snapshot?: any): void {
        if (prevProps.loading && !this.props.loading) {
            Animated.timing(this.state.opacity, {
                toValue: 1,
                duration: 400
            }).start();
        }
    }

    render() {
        return this.props.loading
            ? <Loader loading={true} color={Color.White} style={{justifyContent: 'flex-start'}} />
            : (
                <Animated.View style={[styles.postTabContainer, {opacity: this.state.opacity}]}>
                    {this.props.todos.map(v => (
                        <TitleSubtitleListCell
                            key={v.id}
                            title={v.title}
                            leftAccessory={() => (
                                <Icon
                                    name={v.completed ? "ios-checkmark-circle-outline" : "ios-radio-button-off"}
                                    size={30} />
                            )}
                            cellStyle={[inherited.cell, styles.todoCell]}
                            titleStyle={[inherited.title]}
                            subtitleStyle={inherited.subtitle} />
                    ))}
                </Animated.View>
            );
    }
}

export default withTodosData(UserDetailTodosTab);
