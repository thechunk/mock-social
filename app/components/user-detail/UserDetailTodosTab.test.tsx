import * as React from 'react';
import {UserDetailTodosTab} from './UserDetailTodosTab';

import {shallow} from 'enzyme';
import {Loader} from "../common/loader/index";
import {Animated} from "react-native";

describe('UserDetailTodosTab', () => {
    const todos = [{id: 1, title: 'title', completed: true}];

    it('renders correctly', () => {
        expect(shallow(<UserDetailTodosTab id={1} todos={todos} />)).toMatchSnapshot();
    });

    it('shows loader', () => {
        const wrapper = shallow(<UserDetailTodosTab id={1} loading={true} />);
        expect(wrapper.find(Loader)).toHaveLength(1);
        expect(wrapper.find(Animated.View)).toHaveLength(0);
    });

    it('shows content', () => {
        const wrapper = shallow(<UserDetailTodosTab id={1} loading={false} todos={todos} />);
        expect(wrapper.find(Loader)).toHaveLength(0);
        expect(wrapper.find(Animated.View)).toHaveLength(1);
    })
});
