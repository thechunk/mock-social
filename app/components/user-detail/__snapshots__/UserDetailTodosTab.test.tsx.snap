// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetailTodosTab renders correctly 1`] = `
<AnimatedComponent
  style={
    Array [
      Object {
        "borderTopColor": "rgba(0, 0, 0, 0.2)",
        "borderTopWidth": 0.5,
        "marginTop": 12,
      },
      Object {
        "opacity": 0,
      },
    ]
  }
>
  <TitleSubtitleListCell
    cellStyle={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "minHeight": 48,
          "paddingLeft": 12,
          "paddingRight": 12,
        },
        Object {
          "paddingBottom": 12,
          "paddingTop": 12,
        },
      ]
    }
    key="1"
    leftAccessory={[Function]}
    subtitleStyle={
      Object {
        "fontSize": 12,
      }
    }
    title="title"
    titleStyle={
      Array [
        Object {
          "fontSize": 16,
        },
      ]
    }
  />
</AnimatedComponent>
`;
